### PR TITLE
Add assignment page map handlers

### LINKED ISSUE #1050: Adding Handlers to Assignment page maps
jurisdiction assignment View:
  - can click on a jurisdiction fill to drill down 
both Jurisdiction assignment and plan assignment view:
  - show jurisdiction name on hover.

### LINKED ISSUE #1054: AssigmentMapWrapper changes
- [x] On the JurisdictionAssignment page map, only selected jurisdictions should have a blue border around them
- [x] filterGeom in `JurisdictionGeomFilters` should be true to ensure only jurisdictions with geometry are returned
- [x] Lets add docstrings to `AssignmentMapWrapper`
- [x] `getJurisdictions` should only be called for jurisdictions that we do not yet have in the redux store or jurisdictions in the redux store whose geometry we do not have.  The intention is to limit the requests to the API for jurisdictions.  Please see https://github.com/onaio/reveal-frontend/issues/973#issuecomment-656681240
- [x] add tests to ensure `getJurisdictions` does not get re-called for jurisdictions that are already in the redux store and whose geometry we already have
- [x] In AssignmentMapWrapper `fillColor`, `fillOutlineColor`, and `lineColor` should use colors imported from the colors module and not magic strings
- [x] Consider if indeed `fillColor`, `fillOutlineColor`, and `lineColor` should be properties that are stored to the redux store.  I argue that they should not be stored there because:
    - The reducer module in questions deals with OpenSRP jurisdictions and doe snot really care about mapbox-related properties
    - If these are persisted to redux then whether or not they are re-used across different plans inadvertently becomes unclear.  We should remove this uncertainty
- [x] Reduce [this snapshot](src/containers/pages/AssigmentMapWrapper/tests/__snapshots__/index.test.tsx.snap) size in consultation with https://kentcdodds.com/blog/effective-snapshot-testing/

### PR DESCRIPTION
**Pull request by KipSigei**
Fix #1050  and Fix #1054

### CODE REVIEW

**Comment by KipSigei** [comment]
To Do:

- [ ] Shift + click - should select jurisdiction on map
- [ ] Click - should drill down jurisdictions on map
- [ ] Clicking on jurisdiction checkbox should update layer with blue boundary

File: src/components/page/HeaderBreadcrumb/HeaderBreadcrumb.tsx [hunk 1]
```diff
@@ -12,23 +12,49 @@ export interface Page {
 export interface BreadCrumbProps {
   currentPage: Page;
   pages: Page[];
+  fetchUpdatedCurrentParentHandler: (currentParentId: string, isRootJurisdiction: boolean) => void;
```
Comment (moshthepitt) [inline src/components/page/HeaderBreadcrumb/HeaderBreadcrumb.tsx:15]: Could you describe some more either as a comment or in the PR what this is for?

File: src/containers/pages/AssigmentMapWrapper/index.tsx [hunk 2]
```diff
@@ -32,18 +49,93 @@ export interface AssignmentMapWrapperProps {
...
+  selectNodeCreator: selectNode,
   serviceClass: OpenSRPService,
 };
 
+/**
+ * Handler to get current parent id from clicked jurisdiction on map
+ * @param props - AssignmentMapWrapper component props
+ */
+
+export const onJurisdictionClick = (props: any, setMapParent: any) => {
```
Comment (moshthepitt) [inline src/containers/pages/AssigmentMapWrapper/index.tsx:79]: Lets not use `any` here.

File: src/containers/pages/AssigmentMapWrapper/index.tsx [hunk 3]
```diff
@@ -32,18 +49,93 @@ export interface AssignmentMapWrapperProps {
...
+    const { point, target, originalEvent } = e;
+    const {
+      fetchUpdatedCurrentParentActionCreator,
+      currentParentNode,
+      rootJurisdictionId,
+      selectNodeCreator,
+      deselectNodeCreator,
+    } = props;
+    // grab underlying features from map
+    const features: any = target.queryRenderedFeatures(point) as Dictionary[];
```
Comment (moshthepitt) [inline src/containers/pages/AssigmentMapWrapper/index.tsx:91]: Lets not use `any` here.

File: src/containers/pages/AssigmentMapWrapper/index.tsx [hunk 4]
```diff
@@ -32,18 +49,93 @@ export interface AssignmentMapWrapperProps {
...
+    // dont drill down map if current jurisdiction is leafNode
+    if (!features.length) {
+      return false;
+    }
+    if (features[0]) {
```
Comment (moshthepitt) [inline src/containers/pages/AssigmentMapWrapper/index.tsx:96]: What's the significance of 0 here?  Can we avoid using a [magic number](https://en.wikipedia.org/wiki/Magic_number_(programming)) like this? I'd prefer either a named constant or a map.

Comment (moshthepitt) [reply src/containers/pages/AssigmentMapWrapper/index.tsx:96]: It seems that `features` is an array of something but we are only using the first element.  Why?  Would it make sense to structure it different if we only need one element?

File: src/containers/pages/AssigmentMapWrapper/index.tsx (hunk 3 above)
Comment (moshthepitt) [reply src/containers/pages/AssigmentMapWrapper/index.tsx:91]: Can we use a better defined type than `Dictionary[]`?

File: src/containers/pages/AssigmentMapWrapper/index.tsx (hunk 2 above)
Comment (moshthepitt) [inline src/containers/pages/AssigmentMapWrapper/index.tsx:79]: Can we move this function to a utils or helpers module? Something like `../AssigmentMapWrapper/{utils/helpers}.tsx `

File: src/containers/pages/FocusInvestigation/active/index.tsx [hunk 5]
```diff
@@ -170,7 +170,7 @@ class ActiveFocusInvestigation extends React.Component<
   }
 
   public render() {
-    const breadcrumbProps: BreadCrumbProps = {
+    const breadcrumbProps: Partial<BreadCrumbProps> = {
```
Comment (moshthepitt) [inline src/containers/pages/FocusInvestigation/active/index.tsx:173]: Why is this necessary in this PR?

File: src/containers/pages/JurisdictionAssignment/JurisdictionCell/index.tsx [hunk 6]
```diff
@@ -33,7 +33,12 @@ const NodeCell = (props: NodeCellProps) => {
   }
 
   return (
-    <Link to={`${baseUrl}/${node.model.id}`} key={`${node.model.id}-span`}>
+    <Link
+      to={`${baseUrl}/${node.model.id}`}
+      key={`${node.model.id}-span`}
+      // tslint:disable-next-line: jsx-no-lambda
+      onClick={() => fetchUpdatedCurrentParent('', false)}
```
Comment (moshthepitt) [inline src/containers/pages/JurisdictionAssignment/JurisdictionCell/index.tsx:40]: Why does this have to happen as an onClick event to `Link`?  Can we find another way please?

File: src/containers/pages/JurisdictionAssignment/JurisdictionTable/index.tsx [hunk 7]
```diff
@@ -73,14 +82,16 @@ export interface JurisdictionSelectorTableProps {
...
-  autoSelectNodesCreator: typeof autoSelectNodes;
-  fetchPlanCreator: typeof addPlanDefinition;
+  selectNodeCreator: ActionCreator<SelectNodeAction>;
+  deselectNodeCreator: ActionCreator<DeselectNodeAction>;
+  autoSelectNodesCreator: ActionCreator<AutoSelectNodesAction>;
+  fetchPlanCreator: ActionCreator<AddPlanDefinitionAction>;
+  fetchUpdatedCurrentParentActionCreator: typeof fetchUpdatedCurrentParent;
   selectedLeafNodes: TreeNode[];
-  leafNodes: TreeNode[];
+  leafNodes: TreeNode[] | undefined;
```
Comment (moshthepitt) [inline src/containers/pages/JurisdictionAssignment/JurisdictionTable/index.tsx:91]: This should be `leafNodes?: TreeNode[]`

File: src/containers/pages/JurisdictionAssignment/JurisdictionTable/index.tsx [hunk 8]
```diff
@@ -116,11 +132,17 @@ const JurisdictionTable = (props: JurisdictionSelectorTableProps) => {
...
+    mapCurrentParent,
+    fetchUpdatedCurrentParentActionCreator,
     serviceClass,
     autoSelectionFlow,
     deselectAllNodesCreator,
   } = props;
 
+  function fetchUpdatedCurrentParentHandler(currentParentId: string, isRootJurisdiction: boolean) {
+    fetchUpdatedCurrentParentActionCreator(currentParentId, isRootJurisdiction);
+  }
```
Comment (moshthepitt) [inline src/containers/pages/JurisdictionAssignment/JurisdictionTable/index.tsx:144]: Is it necessary to wrap `fetchUpdatedCurrentParentActionCreator` like this?

File: src/containers/pages/JurisdictionAssignment/JurisdictionTable/index.tsx [hunk 9]
```diff
@@ -169,7 +191,7 @@ const JurisdictionTable = (props: JurisdictionSelectorTableProps) => {
   // been set, if drilling down has begun.
   let derivedParentNode = currentParentNode;
   let derivedChildrenNodes = currentChildren;
-  if (!props.currentParentId) {
+  if (!props.currentParentId && !mapCurrentParent.currentParentId.length) {
```
Comment (moshthepitt) [inline src/containers/pages/JurisdictionAssignment/JurisdictionTable/index.tsx:194]: Why do we need both props.currentParentId and mapCurrentParent.currentParentId?  These should be the same value at all times, no?

Also, I prefer that we destructure `currentParentId` from props

File: src/store/ducks/opensrp/hierarchies/index.tsx [hunk 10]
```diff
@@ -128,6 +146,16 @@ export function fetchTree(
...
+export function fetchUpdatedCurrentParent(
+  currentParentId: string,
+  isRootJurisdiction: boolean
+): FetchUpdatedCurrentParent {
+  return {
+    currentParentId,
+    isRootJurisdiction,
+    type: FETCH_MAP_CURRENT_PARENT,
+  };
+}
```
Comment (moshthepitt) [inline src/store/ducks/opensrp/hierarchies/index.tsx:158]: I am struggling to see why this is necessary.  It seems that we are using this as a way to keep the map's current parent in the redux store.  However, do we not already have that current parent i.e. when you are drilling up and down the JurisdictionTable component, you know the current parent at all times.  Why not just pass that to the map instead of doing this?

File: src/containers/pages/AssigmentMapWrapper/index.tsx (hunk 4 above)
Comment (KipSigei) [reply src/containers/pages/AssigmentMapWrapper/index.tsx:96]: The reason for this is sometimes when features on map are being queried it would return an array of duplicate feature objects

File: src/containers/pages/JurisdictionAssignment/JurisdictionTable/index.tsx (hunk 8 above)
Comment (KipSigei) [reply src/containers/pages/JurisdictionAssignment/JurisdictionTable/index.tsx:144]: This was to avoid having to transform the `HeaderBreadCrumb` component to connect it to store and just have a handler passed to it as a prop to be able to dispatch the `fetchUpdatedCurrentParent` action.

Comment (moshthepitt) [reply src/containers/pages/JurisdictionAssignment/JurisdictionTable/index.tsx:144]: Why do we need to dispatch that action creator when you click on the link?  I would like us to find a way to avoid that.

File: src/store/ducks/opensrp/hierarchies/index.tsx (hunk 10 above)
Comment (KipSigei) [reply src/store/ducks/opensrp/hierarchies/index.tsx:158]: The reason i added this action is that when drilling up and down using the map the we need to know what jurisdiction children need to be rendered next on the table/breacrumb since `currentParentId` will remain the same

File: src/components/GisidaLite/tests/index.test.tsx [hunk 11]
```diff
@@ -10,7 +10,7 @@ describe('components/GisidaLite/arePropsEqual', () => {
   it('returns false if layers length has changed', () => {
     const prevProps: GisidaLiteProps = {
       layers: [],
-      mapCenter: undefined,
+      mapCenter: [10, 10],
```
Comment (moshthepitt) [inline src/components/GisidaLite/tests/index.test.tsx:13]: Does this and line 38 below mean that the map center is by default defined to [10, 10]?  Why?

File: src/containers/pages/AssigmentMapWrapper/helpers/utils.tsx [hunk 12]
```diff
@@ -0,0 +1,110 @@
...
+    if (!features.length) {
+      return false;
+    }
+    if (features[0]) {
+      const currentId =
+        (features[0].id && features[0].id.toString()) ||
+        (features[0] && features[0].properties && features[0].properties.externalId);
+      if (
+        originalEvent.altKey &&
+        PLAN_TYPES_WITH_MULTI_JURISDICTIONS.includes(getInterventionType.valueCodableConcept)
```
Comment (moshthepitt) [inline src/containers/pages/AssigmentMapWrapper/helpers/utils.tsx:59]: At this point do we need to consider that `getInterventionType` might be undefined?

File: src/containers/pages/AssigmentMapWrapper/index.tsx [hunk 13]
```diff
@@ -1,49 +1,97 @@
...
+  features: [],
+  type: 'FeatureCollection',
+};
+
+const defaultProps: AssignmentMapWrapperProps = {
+  autoSelectNodesActionCreator: autoSelectNodes,
+  autoSelectionFlow: false,
   currentChildren: [],
   currentParentId: undefined,
+  currentParentNode: undefined,
```
Comment (moshthepitt) [inline src/containers/pages/AssigmentMapWrapper/index.tsx:74]: I do not think it is necessary to declare `currentParentNode` as undefined, as that is already catered for in the interface.

File: src/containers/pages/FocusInvestigation/map/active/helpers/utils.tsx [hunk 14]
```diff
@@ -480,3 +480,64 @@ export const buildOnClickHandler = (currentPlanId: string) => {
...
+ */
+
+export const buildMouseMoveHandler = () => {
+  function mouseHoverHandler(map: Map, event: EventData) {
+    const popup = new mapboxgl.Popup({
+      closeButton: false,
+    });
+    // Added this part to remove duplicate popup containers showing on map when hovering
+    const nodes: any = document.getElementsByClassName('mapboxgl-popup');
+    let node: any;
```
Comment (moshthepitt) [inline src/containers/pages/FocusInvestigation/map/active/helpers/utils.tsx:497]: Is it possible to get rid of `any` here?

File: src/containers/pages/FocusInvestigation/map/active/helpers/utils.tsx [hunk 15]
```diff
@@ -480,3 +480,64 @@ export const buildOnClickHandler = (currentPlanId: string) => {
...
+    // grab underlying features from map
+    const features: any = event.target.queryRenderedFeatures(event.point) as Dictionary[];
+    //
+    if (!features.length) {
+      return;
+    }
+    // loop through features, setting the name property to content var
+    for (feature of features) {
+      if (feature && feature.properties && feature.properties.name) {
+        content += `<div><center>${feature.properties.name}</center></div>`;
```
Comment (moshthepitt) [inline src/containers/pages/FocusInvestigation/map/active/helpers/utils.tsx:517]: Let's give this div a class name so that in the future if we want to style it we can do so easily.

File: src/store/ducks/opensrp/jurisdictions/index.ts [hunk 16]
```diff
@@ -146,12 +174,64 @@ export const getJurisdictionsArray = () =>
...
+    ],
+    (
+      jurisdictionsArray,
+      newFeatureProps,
+      currentChildren,
+      rootJurisdictionId,
+      planId,
+      metaDataByJurisdiction
+    ): FeatureCollection => {
+      const validFeatures = jurisdictionsArray.filter(item => 'geometry' in item) as Feature[];
```
Comment (moshthepitt) [inline src/store/ducks/opensrp/jurisdictions/index.ts:193]: There's already a selector that does this (`filterGeom`), no?  Why not use it?

File: src/containers/pages/AssigmentMapWrapper/helpers/utils.tsx [hunk 17]
```diff
@@ -0,0 +1,110 @@
...
+      ) {
+        let activeCurrentNode: TreeNode | any = {};
```
Comment (moshthepitt) [inline src/containers/pages/AssigmentMapWrapper/helpers/utils.tsx:61]: Lets not use `any` here.

File: src/containers/pages/AssigmentMapWrapper/helpers/utils.tsx [hunk 18]
```diff
@@ -0,0 +1,110 @@
...
+      deselectNodeCreator,
+      plan,
+      autoSelectionFlow,
+      getJurisdictionsMetadata,
+    } = props;
+
+    const { useContext } = plan;
+    const getInterventionType = useContext.find(
+      context => context.code === 'interventionType'
+    ) as UseContext;
```
Comment (p-netm) [inline src/containers/pages/AssigmentMapWrapper/helpers/utils.tsx:43]: lets use [this](https://github.com/onaio/reveal-frontend/blob/bd679d507b221c883245ad095c144c263811583a/src/helpers/utils.tsx#L882)

File: src/containers/pages/AssigmentMapWrapper/index.tsx [hunk 19]
```diff
@@ -1,49 +1,97 @@
...
+  currentParentId?: string;
+  jurisdictionsChunkSize: number;
+  getJurisdictionsMetadata: Dictionary<Dictionary<Dictionary<Meta>>>;
   currentChildren: TreeNode[];
   serviceClass: typeof OpenSRPService;
   fetchJurisdictionsActionCreator: typeof fetchJurisdictions;
   getJurisdictionsFeatures: FeatureCollection;
+  autoSelectNodesActionCreator: ActionCreator<AutoSelectNodesAction>;
+  selectNodeCreator: ActionCreator<SelectNodeAction>;
+  deselectNodeCreator: ActionCreator<DeselectNodeAction>;
```
Comment (p-netm) [inline src/containers/pages/AssigmentMapWrapper/index.tsx:59]: nitPick: `ActionCreator<Action>` is not as type-safe as `typeof actionCreator`

File: src/containers/pages/AssigmentMapWrapper/index.tsx [hunk 20]
```diff
@@ -52,12 +100,17 @@ const AssignmentMapWrapper = (props: AssignmentMapWrapperProps) => {
     getJurisdictionsFeatures,
     currentParentId,
     rootJurisdictionId,
+    jurisdictionsChunkSize,
   } = props;
   const currentChildIds: string[] = currentChildren.length
```
Comment (p-netm) [inline src/containers/pages/AssigmentMapWrapper/index.tsx:105]: I believe the conditional check for this is unnecessary: `[].map(callback) => []`;

File: src/containers/pages/JurisdictionAssignment/AutoSelectJurisdictions/index.tsx [hunk 21]
```diff
@@ -260,7 +260,10 @@ const mapStateToProps = (
 ): MapStateToProps => {
   const planId = ownProps.match.params.planId;
   const planObj = getPlanDefinitionById(state, planId);
-  const tree = treeByIdSelector(state, { rootJurisdictionId: ownProps.match.params.rootId });
+  const tree = treeByIdSelector(state, {
+    planId,
```
Comment (p-netm) [inline src/containers/pages/JurisdictionAssignment/AutoSelectJurisdictions/index.tsx:264]: why was planId included in this selector's filter.

File: src/containers/pages/JurisdictionAssignment/EntryView/JurisdictionAssignmentReRouting.tsx [hunk 22]
```diff
@@ -144,7 +144,10 @@ const mapStateToProps = (
   ownProps: JurisdictionAssignmentReRoutingProps
 ): MapStateToProps => {
   return {
-    tree: treeSelector(state, { rootJurisdictionId: ownProps.rootJurisdictionId }),
+    tree: treeSelector(state, {
+      planId: ownProps.plan.identifier,
```
Comment (p-netm) [inline src/containers/pages/JurisdictionAssignment/EntryView/JurisdictionAssignmentReRouting.tsx:148]: why was planId included in this selector's filter.

File: src/containers/pages/JurisdictionAssignment/JurisdictionAssignmentView/index.tsx [hunk 23]
```diff
@@ -219,7 +222,7 @@ const mapStateToProps = (
   const planObj = getPlanDefinitionById(state, planId);
   return {
     plan: planObj,
-    tree: treeByIdSelector(state, { rootJurisdictionId: ownProps.match.params.rootId }),
+    tree: treeByIdSelector(state, { rootJurisdictionId: ownProps.match.params.rootId, planId }),
```
Comment (p-netm) [inline src/containers/pages/JurisdictionAssignment/JurisdictionAssignmentView/index.tsx:225]: why was planId included in this selector's filter.

File: src/containers/pages/JurisdictionAssignment/JurisdictionTable/index.tsx [hunk 24]
```diff
@@ -67,18 +75,18 @@ export interface JurisdictionSelectorTableProps {
...
   currentParentNode?: TreeNode;
   currentChildren: TreeNode[];
-  selectNodeCreator: typeof selectNode;
-  deselectNodeCreator: typeof deselectNode;
-  autoSelectNodesCreator: typeof autoSelectNodes;
-  fetchPlanCreator: typeof addPlanDefinition;
+  selectNodeCreator: ActionCreator<SelectNodeAction>;
+  deselectNodeCreator: ActionCreator<DeselectNodeAction>;
+  autoSelectNodesCreator: ActionCreator<AutoSelectNodesAction>;
+  fetchPlanCreator: ActionCreator<AddPlanDefinitionAction>;
```
Comment (p-netm) [inline src/containers/pages/JurisdictionAssignment/JurisdictionTable/index.tsx:88]: `ActionCreator<Action>` is not type-safe as `typeof actionCreator`. You can try removing one of the args from any of the `ActionCreator<Action>`-typed action creators and you will see that no error is thrown(or atleast my editor does not show the red scriggly lines.)

File: src/containers/pages/PlanAssignment/index.tsx [hunk 25]
```diff
@@ -308,7 +308,10 @@ const mapStateToProps = (state: Partial<Store>, ownProps: any): DispatchedStateP
   const organizations = getOrganizationsArray(state);
   const assignments = getAssignmentsArrayByPlanId(state, ownProps.match.params.planId);
 
-  const tree = treeSelector(state, { rootJurisdictionId: ownProps.match.params.planId });
+  const tree = treeSelector(state, {
+    planId: (plan as PlanDefinition).identifier,
```
Comment (p-netm) [inline src/containers/pages/PlanAssignment/index.tsx:312]: why was planId included in this selector's filter.

File: src/store/ducks/opensrp/hierarchies/index.tsx [hunk 26]
```diff
@@ -66,6 +69,12 @@ export interface DeforestAction extends AnyAction {
   treeByRootId: {};
 }
 
+export interface FetchUpdatedCurrentParent extends AnyAction {
```
Comment (p-netm) [inline src/store/ducks/opensrp/hierarchies/index.tsx:72]: is this being used elsewhere?

File: src/store/ducks/opensrp/hierarchies/index.tsx [hunk 27]
```diff
@@ -355,7 +363,7 @@ export interface WithoutMetaFilters {
 /** prop filters to customize selector queries */
 export interface Filters {
   rootJurisdictionId: string /** specify which tree to act upon */;
-  planId?: string /** get selections that are associated with this planId */;
+  planId: string /** get selections that are associated with this planId */;
```
Comment (p-netm) [inline src/store/ducks/opensrp/hierarchies/index.tsx:366]: IMO making planId required will require one to supply it even for selectors like `getTreeById` that does not use it,

File: src/store/ducks/opensrp/hierarchies/index.tsx [hunk 28]
```diff
@@ -547,7 +553,14 @@ export const getCurrentChildren = () => {
  * @param props -  the filterProps
  */
 export const getAllSelectedNodes = () =>
-  createSelector(getTreeWithMeta(), getLeafNodesOnly, computeSelectedNodes);
+  createSelector(
+    getTreeWithMeta(),
```
Comment (p-netm) [inline src/store/ducks/opensrp/hierarchies/index.tsx:557]: heads up:  getTreeWithMeta will be removed.

File: src/containers/pages/AssigmentMapWrapper/helpers/utils.tsx (hunk 12 above)
Comment (moshthepitt) [reply src/containers/pages/AssigmentMapWrapper/helpers/utils.tsx:59]: Actually looks like you should use [this](https://github.com/onaio/reveal-frontend/blob/bd679d507b221c883245ad095c144c263811583a/src/helpers/utils.tsx#L894).

File: src/store/ducks/opensrp/hierarchies/index.tsx (hunk 28 above)
Comment (moshthepitt) [reply src/store/ducks/opensrp/hierarchies/index.tsx:557]: @p-netm how do you propose we resolve that in this PR?

Comment (p-netm) [reply src/store/ducks/opensrp/hierarchies/index.tsx:557]: @moshthepitt by merging #1127

File: src/components/GisidaLite/tests/index.test.tsx (hunk 11 above)
Comment (KipSigei) [reply src/components/GisidaLite/tests/index.test.tsx:13]: I added that value for testing purposes

Comment (moshthepitt) [reply src/components/GisidaLite/tests/index.test.tsx:13]: Cool

**Comment by KipSigei** [comment]
> When i ran this locally, I encountered these issues:
> 
> * can drill down in the map past the operational area.
> * make cursor, pointer if clickable
> * there is no popup for Jurisdiction name on hover => I expected to see this since it looks like there is a mouseMoveHandler added in the pr

@p-netm These have been addressed you can check once more when you get time

File: src/store/ducks/opensrp/hierarchies/index.tsx [hunk 29]
```diff
@@ -55,6 +55,9 @@ export const DESELECT_ALL_NODES = 'opensrp/locations/hierarchy/DESELECT_ALL_NODE
 // /** action to auto-append/ auto-modify the selected attribute of nodes in a tree */
 export const AUTO_SELECT_NODES = 'opensrp/locations/hierarchy/AUTO_SELECT_NODES';
 
+/** action to fetch updated children to ensure map and jurisdiction table are in sync */
+export const FETCH_MAP_CURRENT_PARENT = 'opensrp/locations/hierarchy/FETCH_MAP_CURRENT_PARENT';
```
Comment (p-netm) [inline src/store/ducks/opensrp/hierarchies/index.tsx:59]: is this being used somewhere?

Comment (KipSigei) [reply src/store/ducks/opensrp/hierarchies/index.tsx]: @p-netm  addressed via https://github.com/onaio/reveal-frontend/pull/1083/commits/6ca235ed7f320553ff768af85161ae54f5f39023

File: src/containers/pages/AssigmentMapWrapper/index.tsx [hunk 30]
```diff
@@ -1,49 +1,93 @@
...
+    identifier: '',
+  } as PlanDefinition,
   rootJurisdictionId: '',
+  selectNodeCreator: selectNode,
   serviceClass: OpenSRPService,
 };
 
+/**
+ * This is a map HOC for the plan and jurisdiction assignment pages
+ * It is responsuble for fetching jurisdiction geojson and wiring down to
```
Comment (moshthepitt) [inline src/containers/pages/AssigmentMapWrapper/index.tsx:86]: typo here

File: src/containers/pages/FocusInvestigation/map/active/helpers/utils.tsx [hunk 31]
```diff
@@ -480,3 +480,65 @@ export const buildOnClickHandler = (currentPlanId: string) => {
...
   return clickHandler;
 };
+
+/**
+ * Map component popup handler
+ * @param map - mapbox map instance
+ * @param event - map event object
+ */
+
+export const buildMouseMoveHandler = (map: Map, event: EventData) => {
```
Comment (moshthepitt) [inline src/containers/pages/FocusInvestigation/map/active/helpers/utils.tsx:490]: Let's move this either to GisidaLite utils or the AssignmentMapWrapper utils.  I feel that it doesn't have much to do with FocusInvestigation
